core/types/legacy_tx.go

```diff
@@ -466,3 +476,19 @@ func (tx *LegacyTx) Sender(signer Signer) (libcommon.Address, error) {
 	tx.from.Store(addr)
 	return addr, nil
 }
+
+// The last timestamp check is to ensure that any transaction is not a legacy deposit transaction
+// after the bedrock upgrade.
+func (tx *LegacyTx) IsLegacyDepositTx() bool {
+	V, R, S := tx.RawSignatureValues()
+	// contract creation
+	if tx.To == nil {
+		return false
+	}
+	if *tx.To == MessengerAddress && *V == *uint256.NewInt(0) &&
```

Comment:
No need to create all of these pointers and dereference.  Should be able to simply do:

```*V == (uint256.Int{})```

Reply:
Fixed!